rpc/types_transaction.go

```diff
@@ -43,6 +43,10 @@ type InvokeTxnV0 struct {
 	FunctionCall
 }
 
+func (tx InvokeTxnV0) InvokeType() string {
+	return "InvokeTxn: V0"
+}
+
```

Comment:
We don't need to define this method on the InvokeTxnVX types.

If you are getting errors when trying to pass in (eg) InvokeV1 txns in TestAddInvokeTransaction, then you just need to wrap the InvokeTxnsV1 in BroadcastInvokeV1Txn
`InvokeTx:     BroadcastInvokev1Txn{InvokeTxnV1{SenderAddress: new(felt.Felt).SetUint64(123)}},`

Reply:
Resolved it....
You can have a look at it...

Reply:
Hi @rianhughes 
I did what you told.... But there is a weird error which I'm facing in `rpc/write_test.go` >> `TestAddInvokeTransaction`

**For Mock:**
1. `InvokeTx: BroadcastInvokev3Txn{......`
I got this error 
```
=== RUN   TestAddInvokeTransaction
--- FAIL: TestAddInvokeTransaction (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x88dc32]
```

2. `InvokeTx:  BroadcastInvokev1Txn{InvokeTxnV1{}},`
I got this error 
```
=== RUN   TestAddInvokeTransaction
--- FAIL: TestAddInvokeTransaction (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x88d5f8]
```

3. `InvokeTx:     BroadcastInvokev1Txn{InvokeTxnV1{SenderAddress: ......}}`
I got this error 
```
=== RUN   TestAddInvokeTransaction
    /home/psychopunk_sage/dev/Workplace/Nethermind/starknet.go/rpc/write_test.go:174:
                Error Trace:    /home/psychopunk_sage/dev/Workplace/Nethermind/starknet.go/rpc/write_test.go:174
                Error:          Not equal:
                                expected: &rpc.RPCError{Code:63, Message:"An unexpected error occurred", Data:"Something crazy happened"}
                                actual  : &rpc.RPCError{Code:-32603, Message:"Internal Error", Data:"0  <nil>\n"}
```

Reply:
Like all these errors are unusual... they shouldn't occur, don't know what is the issue (I'm trying to find the solution.)

Reply:
Let me know if you get these errors after trying the new approach as outlined by @thiagodeev in the internal discussions

Reply:
Thanks Its working now....